### PR TITLE
Update Docs for App-Specific Passwords

### PR DESCRIPTION
**Pull request by rcmaehl**
See:
home-assistant/core#101816
home-assistant/core#120945

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
  - Updated iCloud integration instructions to clarify potential for recurring notifications if improperly configured, instructing users to select "Don't Allow" and reconfigure credentials if needed.
  - Emphasized the necessity of using App-Specific Passwords for setting up the integration to enhance security and prevent recurring MFA prompts.

### CODE REVIEW

**Comment by rcmaehl** [comment]
> It seems that this PR is targeted against an incorrect branch since it has a parent PR on one of our codebases. Documentation that needs to be updated for an upcoming release should target the `next` branch. Please change the target branch of this PR to `next` and rebase if needed.

This should not be required as it can apply to the existing configuration without issue. I have updated the title to reflect this.

File: source/_integrations/icloud.markdown [hunk 1]
```diff
@@ -38,7 +38,7 @@ To prevent excessive battery drainage, a dynamic interval is used for each indiv
 
 ## Two Factor Authentication
 
-If two-step authentication is enabled for your iCloud account, some time after Home Assistant startup the integration will ask to enter the verification code you receive on your device via a notification in the Home Assistant UI. The duration of this authentication is determined by Apple, so you will need to verify your account every now and then.
+By default, all users should use [App-Specific Passwords](https://support.apple.com/102654) to prevent recurring MFA prompts. If using the Account Password, and if two-step authentication is enabled for your iCloud account, some time after Home Assistant startup the integration will ask you to enter the verification code you receive on your device via a notification in the Home Assistant UI. The duration of this authentication is determined by Apple, so you will need to verify your account periodically.
```
Comment (frenck) [inline source/_integrations/icloud.markdown]: I think we can just write is short and simple like this:

```suggestion
{% important %}
You need to use should use a [app-specific passwords](https://support.apple.com/102654) for setting up this integration.
{% endimportant %}
```

File: source/_integrations/icloud.markdown [hunk 2]
```diff
@@ -29,7 +29,7 @@ It does require that your devices are registered with the [Find My](https://www.
 {% include integrations/config_flow.md %}
 
 {% warning %}
-You may receive an email and a notification from Apple saying that someone has logged into your account.
+You may receive recurring emails and notifications from Apple stating that someone has logged into your account if the integration is improperly configured.
```
Comment (frenck) [inline source/_integrations/icloud.markdown]: This doesn't help, as it tells the user they did something wrong without providing a solution.

**Review by frenck — DISMISSED** [review]

File: source/_integrations/icloud.markdown (hunk 2 above)
Comment (rcmaehl) [reply source/_integrations/icloud.markdown]: Looking to see if I can figure out a method to update the credentials to App-specific Password for the Integration without removing and readding the Integration.

File: source/_integrations/icloud.markdown [hunk 3]
```diff
@@ -38,7 +38,9 @@ To prevent excessive battery drainage, a dynamic interval is used for each indiv
 
 ## Two Factor Authentication
 
-If two-step authentication is enabled for your iCloud account, some time after Home Assistant startup the integration will ask to enter the verification code you receive on your device via a notification in the Home Assistant UI. The duration of this authentication is determined by Apple, so you will need to verify your account every now and then.
+{% important %}
+You need to use should use a [app-specific passwords](https://support.apple.com/102654) for setting up this integration.
```
Comment (RosemaryOrchard) [inline source/_integrations/icloud.markdown]:
```suggestion
You need to use should use a [app-specific passwords](https://support.apple.com/102654) to set up this integration.
```
Simplify the English.

File: source/_integrations/icloud.markdown (hunk 2 above)
Comment (RosemaryOrchard) [reply source/_integrations/icloud.markdown]: Worst case scenario, could we wait for the current auth period to expire?

Comment (rcmaehl) [reply source/_integrations/icloud.markdown]: I've updated this in 564a43c. Let me know if you'd like the wording changed/improved.

**Comment by frenck** [comment]
@rcmaehl This PR looks pretty ready to go to me, but is still in draft. Are you still planning on making changes? Or did you forget to mark it ready for review maybe?

../Frenck

**Comment by rcmaehl** [comment]
> @rcmaehl This PR looks pretty ready to go to me, but is still in draft. Are you still planning on making changes? Or did you forget to mark it ready for review maybe?
> 
> ../Frenck

Yep. Forgot to mark!

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @rcmaehl :+1:

**Pull request merged**